a3fe/run/leg.py

```diff
@@ -805,31 +803,24 @@ def write_input_files(
                     _shutil.copy(
                         restraint_file, f"{stage_input_dir}/restraint_{i + 1}.txt"
                     )
-
-            # Update the somd.cfg file with the perturbed residue number generated
-            # by BSS, as well as the restraints options
-            
-            # generate the somd.cfg file
-            somd_config = self.engine_config.get_somd_config(
-                run_dir=stage_input_dir,
-                config_name="somd"
-            )
+            # Create a new config for this stage
+            stage_config = self.engine_config.copy()
             
             # Set configuration options
-            somd_config.perturbed_residue_number = perturbed_resnum
-            somd_config.use_boresch_restraints = self.leg_type == _LegType.BOUND
-            somd_config.turn_on_receptor_ligand_restraints_mode = self.leg_type == _LegType.BOUND
-            somd_config.charge_difference = -lig_charge  # Use co-alchemical ion approach when there is a charge difference
+            stage_config.perturbed_residue_number = perturbed_resnum
+            stage_config.use_boresch_restraints = self.leg_type == _LegType.BOUND
+            stage_config.turn_on_receptor_ligand_restraints = self.leg_type == _LegType.BOUND
+            stage_config.charge_difference = -lig_charge  # Use co-alchemical ion approach when there is a charge difference
```

Comment:
Let's just store ligand charge and either just write out charge difference from the ligand charge, or have a charge difference property (first option probably best).